airbyte-db/db-lib/src/main/java/io/airbyte/db/instance/configs/migrations/V0_40_3_003__CreateSyncStats.java

```diff
@@ -0,0 +1,69 @@
+import static org.jooq.impl.DSL.unique;
+
+import java.time.OffsetDateTime;
+import java.util.UUID;
+import org.flywaydb.core.api.migration.BaseJavaMigration;
+import org.flywaydb.core.api.migration.Context;
+import org.jooq.DSLContext;
+import org.jooq.Field;
+import org.jooq.impl.DSL;
+import org.jooq.impl.SQLDataType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class V0_40_3_003__CreateSyncStats extends BaseJavaMigration {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(V0_40_3_003__CreateSyncStats.class);
+
+  @Override
+  public void migrate(final Context context) throws Exception {
+    LOGGER.info("Running migration: {}", this.getClass().getSimpleName());
+
+    // Warning: please do not use any jOOQ generated code to write a migration.
+    // As database schema changes, the generated jOOQ code can be deprecated. So
+    // old migration may not compile if there is any generated code.
+    final DSLContext ctx = DSL.using(context.getConnection());
+    createSyncStatsTable(ctx);
+  }
+
+  private static void createSyncStatsTable(final DSLContext ctx) {
+    final Field<UUID> id = DSL.field("id", SQLDataType.UUID.nullable(false));
+    final Field<Integer> attemptId = DSL.field("attempt_id", SQLDataType.INTEGER.nullable(false));
+    final Field<Integer> recordsEmitted = DSL.field("records_emitted", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> bytesEmitted = DSL.field("bytes_emitted", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> sourceStateMessagesEmitted = DSL.field("source_state_messages_emitted", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> destinationStateMessagesEmitted = DSL.field("destination_state_messages_emitted", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> recordsCommitted = DSL.field("records_committed", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> meanSecondsBeforeSourceStateMessageEmitted =
+        DSL.field("mean_seconds_before_source_state_message_emitted", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> maxSecondsBeforeSourceStateMessageEmitted =
+        DSL.field("max_seconds_before_source_state_message_emitted", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> meanSecondsBetweenStateMessageEmittedandCommitted =
+        DSL.field("mean_seconds_between_state_message_emitted_and_committed", SQLDataType.INTEGER.nullable(true));
+    final Field<Integer> maxSecondsBetweenStateMessageEmittedandCommitted =
+        DSL.field("max_seconds_between_state_message_emitted_and_committed", SQLDataType.INTEGER.nullable(true));
+    final Field<OffsetDateTime> createdAt =
+        DSL.field("created_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false).defaultValue(currentOffsetDateTime()));
+    final Field<OffsetDateTime> updatedAt =
+        DSL.field("updated_at", SQLDataType.TIMESTAMPWITHTIMEZONE.nullable(false).defaultValue(currentOffsetDateTime()));
```

Comment:
Because I don't know - do we expect the default times to be set by the application when a new row is written, or do we want the database to have a default value of `NOW()`?

Reply:
hmm so I was using a default value the way some other migrations are under the assumption that these fields would be generated automatically without the application specifying the current time, but in our actual implementations I'm seeing that we are having the application set the current time for each row. So I'm not sure exactly how the default value works in this case.